### PR TITLE
apprise: vendor Python resources

### PR DESCRIPTION
**Pull request by timsutton**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

Vendor Python resources in `apprise`.

### CODE REVIEW

File: Formula/a/apprise.rb [hunk 1]
```diff
@@ -16,16 +16,24 @@ class Apprise < Formula
...
+    sha256 "ca9853ad459e787e2192211578cc907e7594e294c7ccc834310722b41b9ca6de"
+  end
+
+  resource "pyyaml" do
+    url "https://files.pythonhosted.org/packages/cd/e5/af35f7ea75cf72f2cd079c95ee16797de7cd71f29ea7c68ae5ce7be1eda0/PyYAML-6.0.1.tar.gz"
+    sha256 "bfdf460b1736c775f2ba9f6a92bca30bc2095067b8a9d77876d1fad6cc3b4a43"
+  end
+
   def install
     system python3, "-m", "pip", "install", *std_pip_args, "."
```
Comment (branchvincent) [inline Formula/a/apprise.rb:38]: need to switch to a venv (which also requires an `include Language::Python::Virtualenv`):
```suggestion
    virtualenv_install_with_resources
```

File: Formula/a/apprise.rb [hunk 2]
```diff
@@ -16,16 +16,24 @@ class Apprise < Formula
   end
 
   depends_on "python-setuptools" => :build
-  depends_on "python-click"
   depends_on "python-markdown"
   depends_on "python-requests-oauthlib"
```
Comment (branchvincent) [inline Formula/a/apprise.rb:20]: We can drop these as well

File: Formula/a/apprise.rb (hunk 1 above)
Comment (timsutton) [reply Formula/a/apprise.rb:38]: Good point, had totally forgotten this step

**Comment by bevanjkay** [comment]
@timsutton Does the build succeed without setuptools being vendored?

**Review by branchv — APPROVED** [review]
👍

**Comment by timsutton** [comment]
> @timsutton Does the build succeed with setuptools being vendored?

Thanks, it does! The explanation over [here](https://github.com/Homebrew/homebrew-core/pull/163089#discussion_r1493457732) helped me understand a bit better.

**Pull request merged**